folly/test/FBVectorTest.cpp

```diff
@@ -256,3 +256,15 @@ TEST(FBVector, zero_len) {
   fb6 = il;
   fbvector<int> fb7(fb6.begin(), fb6.end());
 }
+
+#if __cpp_deduction_guides >= 201703 || _MSC_VER
+TEST(FBVector, deduction_guides) {
+  fbvector<int> v(3);
+
+  fbvector x(v.begin(), v.end());
+  static_assert(std::is_same_v<fbvector<int>, decltype(x)>);
+
+  fbvector y{v.begin(), v.end()};
+  static_assert(std::is_same_v<fbvector<fbvector<int>::iterator>, decltype(y)>);
```

Comment:
lol

Reply:
Bit extreme of a test, or laughing because of how the first phase of overload resolution for list-initialization will select the initializer-list constructor?

Reply:
Pay no mind - this is just fine.